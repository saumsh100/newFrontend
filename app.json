  {
  "name": "carecru",
  "scripts": {
  },
  "env": {
    "NODE_ENV": {
      "required": true
    },
    "RETHINKDB_AUTHKEY": {
      "required": true
    },
    "RETHINKDB_DB": {
      "required": true
    },
    "RETHINKDB_HOST": {
      "required": true
    },
    "RETHINKDB_PORT": {
      "required": true
    },
    "VENDASTA_API_KEY": {
      "required": true
    },
    "VENDASTA_API_USER": {
      "required": true
    },
    "TWILIO_ACCOUNT_SID": {
      "required": true
    },
    "TWILIO_AUTH_TOKEN": {
      "required": true
    },
    "TWILIO_NUMBER": {
      "required": true
    }
  },
  "formation": {
    "web": {
      "quantity": 1
    }
  },
  "addons": [

  ],
  "buildpacks": [
    {
      "url": "heroku/nodejs"
    }
  ]
}
