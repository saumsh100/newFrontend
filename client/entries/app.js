
import '../apps/default';
import '../apps/Dashboard';
