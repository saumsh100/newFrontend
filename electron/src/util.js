
module.exports.getChannelFromVersion = (version) => {
  const splitVersion = version.split('-');
  return splitVersion[1] || null;
};
