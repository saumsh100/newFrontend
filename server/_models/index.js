
const Sequelize = require('sequelize');
const { postgres } = require('../config/globals');

const sequelizeConfig = {
  logging: false,
  dialect: 'postgres',
  host: postgres.host,
  port: postgres.port,
  define: {
    underscored: false,
    paranoid: true,
  },
};

// Don't use NODE_ENV===production
// It doesn't let us debug production locally easily
if (postgres.ssl) {
  sequelizeConfig.dialectOptions = {
    ssl: true,
  };
}

// If true, sequelize will dump all PostgreSQL queries into terminal
if (postgres.logging) {
  sequelizeConfig.logging = console.log // eslint-disable-line
}

// initialize sequelize
const sequelize = new Sequelize(
  postgres.database,
  postgres.username,
  postgres.password,
  sequelizeConfig,
  );

// Test the connection for logs
sequelize
  .authenticate()
  .then(() => {
    console.log(`Sequelize connected to database=${postgres.database} at host:port=${postgres.host}:${postgres.port}`);
  })
  .catch(err => {
    console.error('postgres config', postgres);
    console.error('Unable to connect to the database:', err);
  });

// Import and store all models.
const models = [];
// models.push((require('./Address').default(sequelize, Sequelize)));
models.push((require('./Account').default(sequelize, Sequelize)));
models.push((require('./Appointment').default(sequelize, Sequelize)));
models.push((require('./AuthSession').default(sequelize, Sequelize)));
models.push((require('./Call').default(sequelize, Sequelize)));
models.push((require('./Chair').default(sequelize, Sequelize)));
models.push((require('./Chat').default(sequelize, Sequelize)));
models.push((require('./Enterprise').default(sequelize, Sequelize)));
models.push((require('./Family').default(sequelize, Sequelize)));
models.push((require('./Invite').default(sequelize, Sequelize)));
models.push((require('./Patient').default(sequelize, Sequelize)));
models.push((require('./PatientUser').default(sequelize, Sequelize)));
models.push((require('./Permission').default(sequelize, Sequelize)));
models.push((require('./PinCode').default(sequelize, Sequelize)));
models.push((require('./Practitioner').default(sequelize, Sequelize)));
models.push((require('./Practitioner_Service').default(sequelize, Sequelize)));
models.push((require('./PractitionerRecurringTimeOff').default(sequelize, Sequelize)));
models.push((require('./Recall').default(sequelize, Sequelize)));
models.push((require('./Reminder').default(sequelize, Sequelize)));
models.push((require('./Request').default(sequelize, Sequelize)));
models.push((require('./Segment').default(sequelize, Sequelize)));
models.push((require('./SentRecall').default(sequelize, Sequelize)));
models.push((require('./SentReminder').default(sequelize, Sequelize)));
models.push((require('./Service').default(sequelize, Sequelize)));
models.push((require('./SyncClientError').default(sequelize, Sequelize)));
models.push((require('./SyncClientVersion').default(sequelize, Sequelize)));
models.push((require('./TextMessage').default(sequelize, Sequelize)));
models.push((require('./Token').default(sequelize, Sequelize)));
models.push((require('./User').default(sequelize, Sequelize)));
models.push((require('./WaitSpot').default(sequelize, Sequelize)));
models.push((require('./WeeklySchedule').default(sequelize, Sequelize)));

const db = {};
models.forEach((model) => {
  db[model.name] = model;
});

Object.keys(db).forEach((modelName) => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

module.exports = db;
