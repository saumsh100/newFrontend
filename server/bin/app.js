
require('../config/initializeCodeTranspiler');

const globals = require('../config/globals');
const applyWebpack = require('../config/webpack/applyWebpack');
const handleErrors = require('../middleware/handleErrors');
const chalk = require('chalk');
const app = require('../config/express');

// require('../config/kue');

// Uses the NODE_ENV to determine logging full stack or not
app.set('showStackError', true);

// Load in webpack configurations
applyWebpack(app);


// Set the Handlebars templating engine
app.set('views', `${globals.root}/views`);
app.engine('hbs', require('../config/handlebars').engine);
app.set('view engine', 'hbs');

// Extra logging for communication with server
app.use((req, res, next) => {
  console.log(chalk.blue(req.method, req.originalUrl));
  next();
});


// Bind the routes

app.use(require('../routes'));

// Catch errors, log and respond to client
app.use(handleErrors);

// assume 404 since no middleware responded
/*app.use((req, res) => {
  res.status(404).render('404', {
    url: req.originalUrl,
    error: 'Not found',
  });
});*/

module.exports = app;
