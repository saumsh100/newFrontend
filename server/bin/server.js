
const http = require('http');
const globals = require('../config/globals');
const app = require('./app');
const Availability = require('../models/Availability');
const Patient = require('../models/Patient');

const server = http.createServer(app);

const io = require('socket.io')(server);

// TODO: implement Thinky!!!
io.on('connection', (socket) => {
  console.log('a new socket connected');
  socket.on('fetchAvailabilities', () => {
    Availability.run()
      .then((availabilities) => {
        socket.emit('receiveAvailabilities', availabilities);
      });
  });
  
  socket.on('fetchPatients', () => {
    Patient.run()
      .then((patients) => {
        socket.emit('receivePatients', patients);
      });
  });
  
  socket.on('addAvailability', (data) => {
    new Availability(data).save().then((availability) => {
      io.sockets.emit('availabilityAdded', availability);
    });
  });
  
  socket.on('removeAvailability', (data) => {
    Availability.get(data.id).then((availability) => {
      availability.delete()
        .then((result) => {
          io.sockets.emit('availabilityRemoved', result);
        });
    });
  });
  
  socket.on('syncClientTrigger', (data) => {
    console.log('sync client triggers!', 'data:', data);
  });
});

server.listen(globals.port, () => {
  console.log(`CareCru HTTP Server is running on port ${globals.port}`);
});
