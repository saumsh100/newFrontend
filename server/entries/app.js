
import '../../client/apps/default';
import '../../client/apps/Dashboard';
