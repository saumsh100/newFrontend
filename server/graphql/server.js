
import { Router } from 'express';
import graphQLHTTP from 'express-graphql';
import { sequelizeAuthMiddleware } from '../middleware/auth';
import schema from './data/schema';

const GraphQLRouter = Router();

const PORT = 4000;

const GraphQLServer = graphQLHTTP({
  schema,
  graphiql: true,
  pretty: true,
  subscriptionsEndpoint: `ws://localhost:${PORT}/subscriptions`,
});

// Reusing auth middleware as the server is exposed the same way an API would.
GraphQLRouter.all('*', sequelizeAuthMiddleware, GraphQLServer);

export default GraphQLRouter;

