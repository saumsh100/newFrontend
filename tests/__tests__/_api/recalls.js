import request from 'supertest';
import app from '../../../server/bin/app';
import { Account, Recall, Address, Patient, Appointment } from '../../../server/_models';
import wipeModel, { wipeAllModels } from '../../_util/wipeModel';
import { accountId, enterpriseId, seedTestUsers, wipeTestUsers } from '../../_util/seedTestUsers';
import { recallId1, seedTestRecalls } from '../../_util/seedTestRecalls';
import generateToken from '../../_util/generateToken';
import { getModelsArray, omitPropertiesFromBody, omitProperties } from '../../util/selectors';
import { seedTestPatients, patientId } from '../../_util/seedTestPatients';
import { seedTestPractitioners, practitionerId } from '../../_util/seedTestPractitioners';

// TODO: make seeds more modular so we can see here
// const accountId = '1aeab035-b72c-4f7a-ad73-09465cbf5654';
// const patientId = '3aeab035-b72c-4f7a-ad73-09465cbf5654';
const oneDayReminderId = '8aeab035-b72c-4f7a-ad73-09465cbf5654';

const makeApptData = (data = {}) => Object.assign({
  accountId,
  patientId,
  practitionerId,
}, data);

const makePatientData = (data = {}) => Object.assign({
  accountId,
}, data);

const date = (y, m, d, h) => (new Date(y, m, d, h)).toISOString();
const dates = (y, m, d, h) => {
  return {
    startDate: date(y, m, d, h),
    endDate: date(y, m, d, h + 1),
  };
};

const rootUrl = '/_api/accounts';
const accountId2 = '52954241-3652-4792-bae5-5bfed53d37b7';
const addressId = 'd94894b1-84ec-492c-a33e-3f1ad61b9c1c';

const newRecallId = 'f5ab9bc0-f0e6-4538-99ae-2fe7f920abf4';
const address = {
  id: addressId,
  country: 'CA',
  createdAt: '2017-07-19T00:14:30.932Z',
  updatedAt: '2017-07-19T00:14:30.932Z',
};

describe('/api/accounts/:account/recalls', () => {
  // Seed with some standard user data
  let token = null;
  beforeEach(async () => {
    await seedTestUsers();

    // Seed an extra account for fetching multiple and testing switching
    await Address.create(address);

    await Account.create({
      id: accountId2,
      addressId,
      enterpriseId,
      name: 'Test Account 2',
      createdAt: '2017-07-20T00:14:30.932Z',
    });

    await wipeModel(Recall);
    token = await generateToken({ username: 'manager@test.com', password: '!@CityOfBudaTest#$' });
  });

  afterAll(async () => {
    await wipeModel(Recall);
    await wipeTestUsers();
  });

  describe('Recalls', () => {
    beforeEach(async () => {
      await seedTestRecalls();
    });

    describe('GET /:accountId/recalls', () => {
      test('should fetch all recalls for the account', () => {
        return request(app)
          .get(`${rootUrl}/${accountId}/recalls`)
          .set('Authorization', `Bearer ${token}`)
          .expect(200)
          .then(({ body }) => {
            body = omitPropertiesFromBody(body);
            const recalls = getModelsArray('recalls', body);
            expect(recalls.length).toBe(2);
            expect(body).toMatchSnapshot();
          });
      });

      test('should return 403 for different accountIds', () => {
        return request(app)
          .get(`${rootUrl}/${accountId2}/recalls`)
          .set('Authorization', `Bearer ${token}`)
          .expect(403);
      });
    });

    describe('POST /:accountId/recalls', () => {
      test('should create a recall for the account', () => {
        return request(app)
          .post(`${rootUrl}/${accountId}/recalls`)
          .set('Authorization', `Bearer ${token}`)
          .send({
            id: newRecallId,
            primaryTypes: ['sms'],
            interval: '-2 months',
            createdAt: '2017-07-19T00:14:30.932Z',
          })
          .expect(201)
          .then(({ body }) => {
            body = omitPropertiesFromBody(body);
            const recalls = getModelsArray('recalls', body);
            expect(recalls.length).toBe(1);
            expect(body).toMatchSnapshot();
          });
      });

      test('should return 403 for different accountIds', () => {
        return request(app)
          .post(`${rootUrl}/${accountId2}/recalls`)
          .set('Authorization', `Bearer ${token}`)
          .expect(403);
      });
    });

    describe('PUT /:accountId/recalls/:recallId', () => {
      test('should update a recall for the account', () => {
        return request(app)
          .put(`${rootUrl}/${accountId}/recalls/${recallId1}`)
          .set('Authorization', `Bearer ${token}`)
          .send({
            primaryTypes: ['phone'],
            createdAt: '2017-07-19T00:14:30.932Z',
          })
          .expect(200)
          .then(({ body }) => {
            body = omitPropertiesFromBody(body);
            const recalls = getModelsArray('recalls', body);
            const [recall] = recalls;
            expect(recalls.length).toBe(1);
            expect(recall.primaryTypes).toEqual(['phone']);
            expect(body).toMatchSnapshot();
          });
      });

      test('should return 403 for different accountIds', () => {
        return request(app)
          .put(`${rootUrl}/${accountId2}/recalls/${recallId1}`)
          .set('Authorization', `Bearer ${token}`)
          .expect(403);
      });
    });

    describe('DELETE /:accountId/recalls/:recallId', () => {
      afterEach(async () => {
        // have to restore recalls cause these routes could delete
        await seedTestRecalls();
      });

      test('should delete a recall for the account', () => {
        return request(app)
          .delete(`${rootUrl}/${accountId}/recalls/${recallId1}`)
          .set('Authorization', `Bearer ${token}`)
          .expect(204);
      });

      test('should return 403 for different accountIds', () => {
        return request(app)
          .delete(`${rootUrl}/${accountId2}/recalls/${recallId1}`)
          .set('Authorization', `Bearer ${token}`)
          .expect(403);
      });
    });

    describe('GET /:accountId/recalls/outbox', () => {
      beforeEach(async () => {
        await wipeAllModels();
        await seedTestUsers();
        await seedTestPatients();
        await seedTestPractitioners();
        token = await generateToken({ username: 'manager@test.com', password: '!@CityOfBudaTest#$' });

        await Recall.bulkCreate([
          {
            accountId,
            primaryTypes: ['email'],
            interval: '1 months',
          },
        ]);

        const patients = await Patient.bulkCreate([
          makePatientData({ firstName: 'Old', email: 'hello@hello.com', lastName: 'Patient', status: 'Active', lastHygieneDate: date(2016, 7, 5, 9), contCareInterval: '6 months' }),
        ]);

        await Appointment.bulkCreate([
          makeApptData({ patientId: patients[0].id, ...dates(2016, 7, 5, 9) }),
        ]);
      });

      afterAll(async () => {
        await wipeModel(Recall);
        await wipeAllModels();

        await seedTestUsers();
        token = await generateToken({ username: 'manager@test.com', password: '!@CityOfBudaTest#$' });
      });

      test('Outbox for Recalls - Should return Patient Old', () => {
        const startDate = date(2016, 0, 1, 9);
        const endDate = date(2017, 8, 8, 9);
        return request(app)
          .get(`${rootUrl}/${accountId}/recalls/outbox?startDate=${startDate}&endDate=${endDate}`)
          .set('Authorization', `Bearer ${token}`)
          .expect(200)
          .then(async ({ body }) => {
            expect(body[0].patient.lastHygieneDate).toBe(date(2016, 7, 5, 9));

            body[0].patient = omitProperties(body[0].patient, ['id', 'lastHygieneDate']);
            body[0].recall = omitProperties(body[0].recall, ['id']);

            expect(body).toMatchSnapshot();
          });
      });

      test('Outbox for Recalls - Should not return Patient Old as their recall not in range', () => {
        const startDate = date(2016, 0, 1, 9);
        const endDate = date(2016, 8, 8, 9);
        return request(app)
          .get(`${rootUrl}/${accountId}/recalls/outbox?startDate=${startDate}&endDate=${endDate}`)
          .set('Authorization', `Bearer ${token}`)
          .expect(200)
          .then(async ({ body }) => {
            expect(body.length).toBe(0);
          });
      });
    });
  });
});
