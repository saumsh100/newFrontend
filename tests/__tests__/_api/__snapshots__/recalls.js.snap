// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/accounts/:account/recalls Recalls GET /:accountId/recalls should fetch all recalls for the account 1`] = `
Object {
  "entities": Object {
    "recalls": Object {
      "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "id": "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
        "lengthSeconds": null,
        "primaryType": "sms",
      },
      "e5ab9bc0-f0e6-4538-99ae-2fe7f920abf4": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "id": "e5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
        "lengthSeconds": null,
        "primaryType": "sms",
      },
    },
  },
  "result": Array [
    "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
    "e5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
  ],
}
`;

exports[`/api/accounts/:account/recalls Recalls POST /:accountId/recalls should create a recall for the account 1`] = `
Object {
  "entities": Object {
    "recalls": Object {
      "f5ab9bc0-f0e6-4538-99ae-2fe7f920abf4": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "id": "f5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
        "lengthSeconds": null,
        "primaryType": "sms",
      },
    },
  },
  "result": "f5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
}
`;

exports[`/api/accounts/:account/recalls Recalls PUT /:accountId/recalls/:recallId should update a recall for the account 1`] = `
Object {
  "entities": Object {
    "recalls": Object {
      "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "id": "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
        "lengthSeconds": null,
        "primaryType": "phone",
      },
    },
  },
  "result": "d5ab9bc0-f0e6-4538-99ae-2fe7f920abf4",
}
`;
